### PR TITLE
fix: input attributes id, name and for can't be empty

### LINKED ISSUE #2067: [BUG] attributes id, name, for on element input can't be empty
**Describe the bug**
This bug is related to a11y and occurs in components with input like quantity selector

### PR DESCRIPTION
**Pull request by justyna-13**
# Related issue

Closes #2067 

# Scope of work

# Screenshots of visual changes

# Checklist

- [ ] No commented blocks of code left
- [ ] Run tests and docs
- [ ] Self code-reviewed
- [ ] Changes documented 

If applicable:

- [ ] I followed [composition rules](https://docs.storefrontui.io/?path=/story/introduction-contributing-guide-code-guidelines--page) for my component
- [ ] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/vue/src/components/atoms/SfQuantitySelector/SfQuantitySelector.vue [hunk 1]
```diff
@@ -24,6 +24,7 @@
     <component
       :is="injections.components.SfInput"
       type="number"
+      :name="'quantitySelector' + $options.uniqueKey()"
```
Comment (aniamusial) [inline packages/vue/src/components/atoms/SfQuantitySelector/SfQuantitySelector.vue:27]: suggestion: I would move it to a computed property :) fe. inside uniqueKey so it returns 'quantitySelector' + key

**Review by aniamusial — APPROVED** [review]

File: packages/vue/src/components/atoms/SfQuantitySelector/SfQuantitySelector.vue [hunk 2]
```diff
@@ -99,6 +100,10 @@ export default {
   handleBlur(listeners) {
     return listeners.blur && listeners.blur();
   },
+  uniqueKey() {
+    const key = Math.random().toString(16).slice(2);
+    return "quantitySelector" + key;
```
Comment (AdamPawlinski) [inline packages/vue/src/components/atoms/SfQuantitySelector/SfQuantitySelector.vue:105]: I would add hyphen between quantitySelector and key number for greater readability, wdyt?

File: packages/vue/src/components/atoms/SfInput/SfInput.vue [hunk 3]
```diff
@@ -129,6 +129,9 @@ export default {
     isPassword() {
       return this.type === "password" && this.hasShowPassword;
     },
+    idWithoutWhitespace() {
+      return this.name.replace(/\s/g, "");
```
Comment (AdamPawlinski) [inline packages/vue/src/components/atoms/SfInput/SfInput.vue:133]: Wdyt about changing the name to PascalCase or camelCase convetion?

**Review by AdamPawlinski — COMMENTED** [review]
It's great work, especially for functional component. I added two suggestions, but it works so they are not necessary to implement :)

**Pull request merged**